istioctl/cmd/istioctl/main.go

```diff
@@ -206,9 +206,10 @@ See https://istio.io/docs/reference/ for an overview of Istio routing.
 				// fill up revision
 				if config.ResourceVersion == "" {
 					current, exists := configClient.Get(config.Type, config.Name, config.Namespace)
-					if exists {
-						config.ResourceVersion = current.ResourceVersion
```

Comment:
There is no connected issue.  Is the purpose of this to change the behavior of `istioctl replace`?  The error message may not be clear to CLI users who don't know that Kubernetes CRDs are expected to have revisions.  Let's have a message whose meaning will make sense to someone incorrectly using `istioctl replace`.

Reply:
Hi, @esnible .

The PR may not change the behavior of istioctl replace. 

As code in function ``func (cl *Client) Update(config model.Config)`` of ``istio/pilot/pkg/config/kube/crd/client.go`` in line 422, if ``config.ResourceVersion`` is empty, the function will also return this message. So we can return this error before calling ``func (cl *Client) Update(config model.Config)`` function.

```
        ...
	if config.ResourceVersion == "" {
		return "", fmt.Errorf("revision is required")
	}
        ...
```

Early return may make code more clearly and faster.

Reply:
Hi @esnible,  is there something wrong with the commit?

Reply:
Under what circumstance will the error "revision is required" be surfaced in the CLI?
The message "revision is required" will *never* make sense to an end user.  I am requesting that all CLI messages make sense to the CLI user.

Reply:
OK, I turn off the PR.